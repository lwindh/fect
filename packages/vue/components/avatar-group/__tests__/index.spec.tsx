import { mount } from '@vue/test-utils'
import AvatarGroup from '../index'
import Avatar from '../../avatar'

describe('AvatarGroup', () => {
  it('should be work as avatar container', () => {
    const wrapper = mount({
      render() {
        return (
          <AvatarGroup count="10">
            <Avatar text="a" stacked />
            <Avatar text="b" stacked />
          </AvatarGroup>
        )
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    expect(wrapper.find('.fect-avatar__counter').text()).toBe('+10')

    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('avatar context props should be work', () => {
    const wrapper = mount({
      render() {
        return (
          <AvatarGroup count="10" stacked size="mini" isSquare>
            <Avatar text="a" />
            <Avatar text="b" isSquare={false} />
          </AvatarGroup>
        )
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
