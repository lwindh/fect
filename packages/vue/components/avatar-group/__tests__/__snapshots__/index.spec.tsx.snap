// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup avatar context props should be work 1`] = `
<div class="fect-avatar__group">
  <div class="fect-avatar mini isSquare stacked ">
    <!----><span class="fect-avatar-text">a</span>
  </div>
  <div class="fect-avatar mini isSquare stacked ">
    <!----><span class="fect-avatar-text">b</span>
  </div><span class="fect-avatar__counter">+10</span>
</div>
`;

exports[`AvatarGroup should be work as avatar container 1`] = `
<div class="fect-avatar__group">
  <div class="fect-avatar medium ">
    <!----><span class="fect-avatar-text">a</span>
  </div>
  <div class="fect-avatar medium ">
    <!----><span class="fect-avatar-text">b</span>
  </div><span class="fect-avatar__counter">+10</span>
</div>
`;
